scripts/gen_syscall_header.py

```diff
@@ -40,7 +40,32 @@ def gen_fn(ret, argc, name, extern=False):
                 sys.stdout.write(", ")
     sys.stdout.write(")")
 
-def gen_make_syscall(ret, argc):
+def tabs(count):
+    sys.stdout.write("\t" * count);
+
+def gen_make_syscall(ret, argc, tabcount):
+    tabs(tabcount)
+
+    # The core kernel is built with the --no-whole-archive linker option.
+    # For all the individual .o files which make up the kernel, if there
+    # are no external references to symbols within these object files,
+    # everything in the object file is dropped.
+    #
+    # This has a subtle interaction with system call handlers. If an object
+    # file has system call handler inside it, and nothing else in the
+    # object file is referenced, then the linker will prefer the weak
+    # version of the handler in the generated syscall_dispatch.c. The
+    # user will get an "unimplemented system call" error if the associated
+    # system call for that handler is made.
+    #
+    # Fix this by making a fake reference to the handler function at the
+    # system call site. The address gets stored inside a special section
+    # "hndlr_ref".  This is enough to prevent the handlers from being
+    # dropped, and the hndlr_ref section is itself dropped from the binary
+    # from gc-sections; these references will not consume space.
+
+    sys.stdout.write("static _GENERIC_SECTION(hndlr_ref) __used void *href = &_handler_##name; \\\n")
```

Comment:
This seems to be causing problems on some targets.  Instead of declaring a `void *`, declare a `u32_t (*href)()` instead.

Reply:
This is a C++ issue, not sure if that will specifically fix it but I'll address this problem